daemon/cmd/daemon_main.go

```diff
@@ -102,6 +104,8 @@ var (
 
 	bootstrapTimestamp = time.Now()
 
+	needsRelaxVerifierKernelVer = versioncheck.MustCompile("<5.3.0")
```

Comment:
We usually probe for kernel features using `probe.NewProbeManager()` (see `bpf_sk_assign` in the same file). That relies on `bpftool feature probe macro`. If the feature we need isn't exposed by bpftool, then we usually rely on some helper introduced around the same time and add the wanted feature probe to https://github.com/cilium/cilium/issues/11014.

In your case, maybe `HAVE_LARGE_INSN_LIMIT` is a good fit? IIRC, that was introduced around the same time, maybe even in the same batch?

Reply:
Yup, I exposed the `have_large_insn_limit` feature which tests the presence of the `c04c0d2b968a ("bpf: increase complexity limit and maximum program size")` commit :+1: